### DISCUSSION TITLE
Kiosk , added to Home Assistant via HACS, not working

### DISCUSSION THREAD

**Discussion by theromych** [NemesisRE/kiosk-mode#215] [General]
Hi All
I am having some trouble with configuring Kiosk in Home Assistant
I installed Kiosk via HACS and added

> resources:
>   - url: /hacsfiles/kiosk-mode/kiosk-mode.js
>     type: module

in configuration.yaml file as per instructions, but getting following error in a log after reboot

> 2024-03-21 20:07:36.617 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
> 2024-03-21 20:07:48.026 ERROR (MainThread) [homeassistant.setup] Setup failed for ‘resources’: Integration not found.

Any idea how to sort it?
Thanks

**Comment by elchininet** [comment]
Hi @theromych,
We need to change the documentation, that configuration has been deprecated. Follow these instructions that are practically the same as kiosk-mode: https://github.com/elchininet/keep-texts-in-tabs?tab=readme-ov-file#if-you-are-in-yaml-mode
Let me know if it solves your issue.
Regards

**Reply by elchininet** [reply]
By the way, you need to do that only if you are in `yaml` mode, users using the default `storage` mode should get `kiosk-mode` by default after installing it through HACS.

**Reply by theromych** [reply]
And haw is it "practically the same as kiosk-mode"?

**Reply by elchininet** [reply]
@theromych,

I mean the instructions to install the plugin, not practically the same, they are the same. Specifically this part:

<img width="857" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/599769cb-d029-456d-a105-a3ba9861b7f7">

**Reply by theromych** [reply]
Oh, got ya
 I just replaced 
> resources:
>   url: /hacsfiles/kiosk-mode/kiosk-mode.js
>   type: module

with

> frontend:
>  extra_module_url: /hacsfiles/kiosk-mode/kiosk-mode.js

and it works now :))

Thanks a lot!

**Reply by elchininet** [reply]
Super, thanks for confirming.
I'll submit a PR to change the old instructions.
Regards

**Comment by daviddavid** [comment]
@elchininet I'm having trouble with this as well. I installed Kiosk Mode through HACS, updated the configuration.yaml with the `extra_module_url` pointing to the js file, and added `kiosk: true` to the dashboard's configuration:

![dashboard config](https://github.com/NemesisRE/kiosk-mode/assets/739378/9b293185-b38f-495c-be95-216b5779e565)

Using a non-admin account, I see no difference. I tried removing the `non_admin_settings:` line, logging out, and reloading HA... but it never takes effect. I can see the JS file being loaded in my network tab, though.

**Reply by elchininet** [reply]
Hi @daviddavid,

If you install the plugin through HACS and you are in storage mode, you don‘t need to do anything with the `extra_module_url`, [this is only needed if you are in `yaml` mode](https://github.com/NemesisRE/kiosk-mode/discussions/215#discussioncomment-8878202). Are you in `yaml` mode?

But anyway, that should not be related to your issue, so let‘s try to debug it. 

First of all, check on your console that the `kiosk_mode` console info is there and that there are no erros, then we can continue the debugging of your issuse.

Regards

**Reply by daviddavid** [reply]
> Are you in yaml mode?

I just got started with HA, so I guess not, since that's not the default mode, right? I have HA running in a docker container and added the `extra_module_url` to the configuration.yaml there.

> First of all, check on your console that the kiosk_mode console info is there and that there are no erros, then we can continue the debugging of your issuse.

Yeah, I can see the kiosk-mode info in the console, showing me it's on version 6.0.0.

**Reply by elchininet** [reply]
>I just got started with HA, so I guess not, since that's not the default mode, right? I have HA running in a docker container and added the extra_module_url to the configuration.yaml there.

You don't need to add the `extra_module_url` then. If you didn't follow [these steps](https://www.home-assistant.io/dashboards/dashboards/#using-yaml-for-the-overview-dashboard) you are in storage mode.

>Yeah, I can see the kiosk-mode info in the console, showing me it's on version 6.0.0.

Any error in the console?

If no errors, just add this to the config:

```yaml
kiosk_mode:
  hide_account: true
```

And check if the account avatar gets hidden (number 5 in this screen):

![image](https://github.com/NemesisRE/kiosk-mode/assets/3728220/984dadc9-34c5-4296-8f89-ebf20b4b96cd)

**Reply by daviddavid** [reply]
Ohhh... this only works on **controlled** dashboards. I was trying it out on the main Overview dashboard, which is still set to automatic. HA sure does have its quirks.

Is there a reason kiosk mode doesn't work on non-controlled dashboards? Not that I need it to, just curious.

**Reply by elchininet** [reply]
Hi @daviddavid,

Kiosk mode works on dashboards in which you have configured it. As it is stated in the [important info](https://github.com/NemesisRE/kiosk-mode?tab=readme-ov-file#important-info):

* Config is placed in the root of your Lovelace config: kiosk_mode: should not be indented & **is per dashboard**.
* **If you want the same settings on other dashboards you'll need to repeat the config on those dashboards as well**.

But you can experiment with the query string caching to see if it first your needs:

https://github.com/NemesisRE/kiosk-mode?tab=readme-ov-file#query-string-caching

Regards

**Reply by daviddavid** [reply]
But that's not the case. If I add the `kiosk_mode` line(s) to a non-controlled dashboard, it doesn't work. If I add it to a controlled dashboard, it does what it's supposed to. I would definitely mention that in the README – or consider it a bug, if it's supposed to work with all dashboards that include the `kiosk_mode` config. Happy to help debug this further, if it's a bug.

**Reply by elchininet** [reply]
Hi @daviddavid,

If that is happening it should be something new from the latest versions of Home Assistant but for sure `kiosk-mode` worked in the past on all lovelace dashboards independiently if they were controlled or not.

Most probably something has changed and what one configures in those dashboards is not being exposed publicly anymore. I‘ll check it when I get some time and if that is the case, I‘ll add it to the documentation.

Could you try to check if the url parameters don't work too?

Regards

**Reply by daviddavid** [reply]
> Could you try to check if the url parameters don't work too?

Yea, I checked those before. The cache clearing one (`?clear_km_cache`) didn't fix the non-controlled dashboard issue. But `?disable_km` did work, when the config worked on controlled dashboards.

**Reply by elchininet** [reply]
I am testing it with an image from May and I was wrong, what happened before was that it was impossible to acces the configuration of an uncontrolled dashboard, if one cliked on "Edit dashboard", this appeared:

<img width="664" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/e6227eb4-cd34-4877-8ee4-3da885b21855">

So, to add `kiosk-mode` config, one needed to click on `Take control` and then it converted that dashboard in a controlled one.

At the moment, with `2024.4.1`, if one clicks on "Edit dashboard" what appears is this that is new to me:

<img width="440" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/ea01c92e-aaa4-47ec-aabe-94c6b55bb8d4">

Here, if one goes to the raw configuration editor and add something there:

<img width="341" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/91fe8af7-e0f9-4058-b3dc-f78513f73eec">

This is just ignored, so there is no way of accessing the configuration:

<img width="324" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/89e73e3d-6dc2-4c3a-8052-fc94dc5db47e">

I‘ll submit a PR to change the documentation stating these new changes.

Regards

**Reply by elchininet** [reply]
This change was introduced in `2023.12.0`:

https://www.home-assistant.io/blog/2023/12/06/release-202312/#options-for-the-default-dashboard

**Comment by elchininet** [comment]
>Yea, I checked those before. The cache clearing one (?clear_km_cache) didn't fix the non-controlled dashboard issue. But ?disable_km did work, when the config worked on controlled dashboards.

I mean [the url query strings](https://github.com/NemesisRE/kiosk-mode?tab=readme-ov-file#query-strings). All the options have a URL query counterpart. I can confirm that url query strings still work in these kinds of dashboards.

<img width="1354" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/ce56c8c1-fb3a-4c4c-a451-cebea4b5b9f9">

**Reply by daviddavid** [reply]
Ok, yeah, those work for me as well on a non-controlled dashboard.

**Reply by elchininet** [reply]
Perfect.
Thanks for the heads up, I‘ll update the documentation with these new changes. 👍

**Comment by albertusQuelle** [comment]
cant get it running :( same as the writer before
<img width="914" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/2970225/e2ae2818-33f7-4317-b79a-c589c82c685e">

**Reply by elchininet** [reply]
Hi @lordvalium,
The issue with the writer before was because an uncontrolled dashboard and that seems to be your case looking at your screenshot. That is already stated in the [Important Info](https://github.com/NemesisRE/kiosk-mode?tab=readme-ov-file#important-info) section in the documentation:

>Since Home Assistant `2023.12.0` it is possible to access the raw config of a "Default dashboard" controlled by Home Assistant. It is important to remark that any configuration that you add to these kinds of dashboards will be ignored so `kiosk-mode` will not be able to read it. You need to create a "New dashboard from scratch" controlled by you to make it work with `kiosk-mode`.

**Reply by elchininet** [reply]
Just in case that you have a dashboard controlled by you, the way to access the raw configuration is also stated in the documentation:

>Non-YAML users (Storage Mode) need to add the configuration to each lovelace panel going to `Edit Dashboard` option (located in the overflow menu that appears when one clicks on the top-right three-dots button). Once in `Edit Dashboard` mode, click again on the top-right three-dots button and select `Raw configuration editor`.

You should get a screenshot like the next one:

<img width="1725" alt="image" src="https://github.com/NemesisRE/kiosk-mode/assets/3728220/3e4c8fd8-3968-4206-b0ca-f56d66110e18">

**Reply by albertusQuelle** [reply]
I have created a brand new one but its not working unfortuntely

**Reply by elchininet** [reply]
@lordvalium,
Then check, the steps to edit the configuration. Taking into account your screenshot, it seems that you are editing a view instead of the raw configuration of the dashboard.

**Reply by albertusQuelle** [reply]
Hmm that can be. Then i dont know what i can do or how the steps would be

**Reply by elchininet** [reply]
@lordvalium,
I left the steps in a previous comment and they are also documented in the README. Try them out and let me know.
Regards

**Reply by elchininet** [reply]
Hi @lordvalium,
Did you solve your issue?

**Reply by albertusQuelle** [reply]
Hi, 
Unfortunately no.
 Related a Complete new dashboard.

**Reply by elchininet** [reply]
@lordvalium,
Please, open an issue filling all the details and provide a detailed description of the steps that you have followed. Some screenshots or a video will help the most. 
Regards
